Backend/BlockChangesService.cs

```diff
@@ -25,7 +24,7 @@ public async Task AddAsync(BlockChangeRequest entityToAdd)
             await _blockChangesCollection.InsertOneAsync(entityToAdd);
         }
 
-        public async Task<BlockChangeRequest> FindExchange(BlockChangeRequest blockRequest)
+        private async Task<BlockChangeRequest> FindExchange(BlockChangeRequest blockRequest)
```

Comment:
What if teher is more possible results? may be you should sort them by date of creation and then select the first one.

Reply:
ok I will sort it